contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -697,21 +718,45 @@ Datum
 varbinaryvarchar(PG_FUNCTION_ARGS)
 {
 	bytea	   *source = PG_GETARG_BYTEA_PP(0);
-	char	   *data = VARDATA_ANY(source);
+	char	   *data = VARDATA_ANY(source);		/* Source data is server encoded */
+	VarChar    *result;
+	char 	   *encoded_result;
 	size_t		len = VARSIZE_ANY_EXHDR(source);
 	int32		typmod = PG_GETARG_INT32(1);
 	int32		maxlen = typmod - VARHDRSZ;
-	VarChar    *result;
+	coll_info	collInfo;
+	int			encodedByteLen;
 
 	/*
-	 * Cast the entire input binary data if maxlen is invalid or supplied data
-	 * fits it
+	 * Allow trailing null bytes 
+	 * Its safe since multi byte UTF-8 does not contain 0x00 
```

Comment:
make this more clearer that it is applicable for binary --> var/char cast. And add example also in comment